rust-runtime/aws-smithy-runtime/src/client/defaults.rs

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Runtime plugins that provide defaults for clients.
+//!
+//! Note: these are the absolute base-level defaults. They may not be the defaults
+//! for _your_ client, since many things can change these defaults on the way to
+//! code generating and constructing a full client.
+
+use crate::client::retries::strategy::StandardRetryStrategy;
+use crate::client::retries::RetryPartition;
+use aws_smithy_async::rt::sleep::default_async_sleep;
+use aws_smithy_async::time::SystemTimeSource;
+use aws_smithy_runtime_api::client::http::SharedHttpClient;
+use aws_smithy_runtime_api::client::runtime_components::RuntimeComponentsBuilder;
+use aws_smithy_runtime_api::client::runtime_plugin::{
+    Order, SharedRuntimePlugin, StaticRuntimePlugin,
+};
+use aws_smithy_runtime_api::shared::IntoShared;
+use aws_smithy_types::config_bag::Layer;
+use aws_smithy_types::retry::RetryConfig;
+use aws_smithy_types::timeout::TimeoutConfig;
+use std::borrow::Cow;
+
+/// Runtime plugin that provides a default connector.
+pub fn default_http_client_plugin() -> SharedRuntimePlugin {
+    let _default: Option<SharedHttpClient> = None;
+    #[cfg(feature = "connector-hyper-0-14-x")]
+    let _default = crate::client::http::hyper_014::default_client();
+
+    StaticRuntimePlugin::new()
+        .with_order(Order::Defaults)
+        .with_runtime_components(
+            RuntimeComponentsBuilder::new("default_http_client_plugin").with_http_client(_default),
```

Comment:
I'd consider making this return `Option<Plugin>` instead—it may be a little confusing to have a runtime plugin configured that _doesn't set an HTTP client.

I'd also name the plugin to clarify that it's setting a hyper client.

Reply:
Fixed in ea3ae9620768fb652dbffc9c7eeef07cccfc3833